tests/h/services/oauth_validator_test.py

```diff
@@ -304,10 +323,37 @@ def test_it_sets_expires(self, svc, token_payload, oauth_request, utcnow):
         token = svc.save_bearer_token(token_payload, oauth_request)
         assert token.expires == datetime.datetime(2017, 7, 13, 19, 29, 28)
 
+    def test_it_sets_refresh_token_expires(self, svc, token_payload, oauth_request, utcnow):
+        utcnow.return_value = datetime.datetime(2017, 7, 13, 18, 29, 28)
+
+        token = svc.save_bearer_token(token_payload, oauth_request)
+        assert token.refresh_token_expires == datetime.datetime(2017, 7, 13, 20, 29, 28)
+
     def test_it_sets_authclient(self, svc, token_payload, oauth_request):
         token = svc.save_bearer_token(token_payload, oauth_request)
         assert token.refresh_token == 'test-refresh-token'
 
+    def test_it_removes_refresh_token_expires_in_from_payload(self, svc, token_payload, oauth_request):
+        assert 'refresh_token_expires_in' in token_payload
+        svc.save_bearer_token(token_payload, oauth_request)
+        assert 'refresh_token_expires_in' not in token_payload
+
+    def test_it_invalidates_old_refresh_token(self, svc, token_payload, oauth_request, patch):
+        invalidate_refresh_token = patch('h.services.oauth_validator.OAuthValidatorService.invalidate_refresh_token')
```

Comment:
Does the `patch` call here need to be undone in some way when the test completes, or if it fails for any reason?